Parsers/ASimWebSession/Parsers/vimWebSessionF5ASM.yaml

```diff
@@ -0,0 +1,261 @@
+                                    "Both",
+                                    temp_SrcMatch,
+                                    "SrcIpAddr",
+                                    temp_DstMatch1,
+                                    "DstIpAddr",
+                                    "No match"
+                                )
+          | where ASimMatchingIpAddr != "No match"
+          | parse-kv DeviceCustomString3 as (Host: string, ["User-Agent"]: string, Cookie: string, Referer: string) with (pair_delimiter="\\r\\n", kv_delimiter=":")
+          | parse DeviceCustomString3 with * "HTTP/" HttpVersion: string "\\r\\n" rest: string
+          | extend
+              EventResultDetails = tostring(FieldDeviceCustomNumber1)
+          | project-rename 
+              DstIpAddr = DestinationIP,
+              DstPortNumber = DestinationPort,
+              EventOriginalUid = ExtID,
+              HttpRequestMethod = RequestMethod,
+              NetworkApplicationProtocol = ApplicationProtocol,
+              HttpCookie = Cookie,
+              HttpHost = Host,
+              HttpReferrer = Referer,
+              HttpUserAgent = ['User-Agent'],
+              HttpRequestXff = DeviceCustomString5
+          | extend
+              EventResult = iff(toint(EventResultDetails) >= 400 or DeviceAction =~ "blocked", "Failure", "Success")
+          | where eventresult=='*' or EventResult =~ eventresult
+          | extend
+              HttpStatusCode = EventResultDetails,
+              AdditionalFields = bag_pack(
+                        "Full Request", DeviceCustomString3, 
+                        "Policy Name", DeviceCustomString1,
+                        "Attack Type", DeviceCustomString4,
+                        "Policy Apply Date", DeviceCustomDate1
+                    );
+      let AnomalyDetectionData = AllData
+          | where DeviceEventClassID in ("Brute Force Attack", "IP Enforcer Attack", "Web Scraping Attack", "DoS Attack")
+          | where array_length(httpuseragent_has_any) == 0 
+          | where array_length(eventresultdetails_in) == 0
+          | extend temp_DstMatch2 = has_any_ipv4_prefix(DvcIpAddr, ipaddr_has_any_prefix)
+          | extend ASimMatchingIpAddr = case(
+                                    array_length(src_or_any) == 0,
+                                    "-",
+                                    temp_SrcMatch and temp_DstMatch2,
+                                    "Both",
+                                    temp_SrcMatch,
+                                    "SrcIpAddr",
+                                    temp_DstMatch2,
+                                    "DstIpAddr",
+                                    "No match"
+                                )
+          | where ASimMatchingIpAddr != "No match"
+          | extend
+              EventResult = iff(DeviceAction =~ "passed", "Success", "Failure")
+          | where eventresult=='*' or EventResult =~ eventresult
```

Comment:
As in both code blocks, the EventResult field is derived from the different condition we have added that filters separately in each block after the derivation of the EventResult field.
In case of the ASimMatchingIpAddr field, We have different mapping for DstIpAddr hence added in separate block.